#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 100, 0, 400)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 40, 0, 100)
gen.add("rotational_damping_factor", double_t, 0, "Cartesian rotational damping factor", 1.0, 0.0, 1.0)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 2.0, 0, 100)
gen.add("filter_coeff", double_t, 0, "Filter coef for interpolation", 0.005, 0, 1)
gen.add("q_kp_0", double_t, 0, "Joint0 kp for velocity to torque error", 8, 0, 25)
gen.add("q_kp_1", double_t, 0, "Joint1 kp for velocity to torque error", 7, 0, 20)
gen.add("q_kp_2", double_t, 0, "Joint2 kp for velocity to torque error", 6, 0, 20)
gen.add("q_kp_3", double_t, 0, "Joint3 kp for velocity to torque error", 4, 0, 10)
gen.add("q_kp_4", double_t, 0, "Joint4 kp for velocity to torque error", 2, 0, 6)
gen.add("q_kp_5", double_t, 0, "Joint5 kp for velocity to torque error", 1.5, 0, 6)
gen.add("q_kp_6", double_t, 0, "Joint6 kp for velocity to torque error", 2.5, 0, 6)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
